lib/helpers/query/castUpdate.js

```diff
@@ -42,6 +41,11 @@ module.exports = function castUpdate(schema, obj, options, context, filter) {
     return obj;
   }
 
+  if (schema.options.strict === 'throw') {
+    throw new StrictModeError();
+  } else if (!('overwriteDiscriminatorKey' in context._mongooseOptions)) {
+    delete obj.type;
```

Comment:
This needs to be `delete obj[schema.options.discriminatorKey]` because the user can set the `discriminatorKey` option to whatever they want, it won't always be `type`